<launch>
  <arg name="input_cmd" value="0" />        <!-- Set input mode (0 velocity/steer / 1 force/steer) -->
  <arg name="actuator_model" value="1" />   <!-- Set actuator model (0 ideal / 1 real) -->
  <arg name="tyre_model" value="1" />       <!-- Set tyre model (0 linear / 1 Fiala with saturation / 2 Fiala without saturation) -->

  <!--- Run single track simulator node -->
  <node pkg="singletrack_sim" type="singletrack_vxvy_sim" respawn="false" name="singletrack_vxvy_sim" output="screen">
    <rosparam file="$(find singletrack_sim)/config/singletrack_vxvy_sim.yaml" command="load" />
    <rosparam param="/use_sim_time">true</rosparam>
    <rosparam param="/singletrack_vxvy_sim/input_cmd" subst_value="True">$(arg input_cmd)</rosparam>
    <rosparam param="/singletrack_vxvy_sim/actuator_model" subst_value="True">$(arg actuator_model)</rosparam>
    <rosparam param="/singletrack_vxvy_sim/tyre_model" subst_value="True">$(arg tyre_model)</rosparam>
    <rosparam param="/singletrack_vxvy_sim/automode_delay">0.0</rosparam> <!-- Set delay before starting autonomous mode -->
    <rosparam param="/singletrack_vxvy_sim/Vx0">1.0</rosparam> <!-- Set initial longitudinal velocity -->
    <rosparam param="/singletrack_vxvy_sim/Vy0">0.0</rosparam> <!-- Set initial lateral velocity -->
  </node> 

  <!--- Run test single track simulator node -->
  <node pkg="singletrack_sim" type="test_singletrack_sim" respawn="false" name="test_singletrack_sim" output="screen">
    <rosparam param="/test_singletrack_sim/input_cmd" subst_value="True">$(arg input_cmd)</rosparam>
  </node> 
</launch>
